frontend/src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/index.tsx

```diff
@@ -1,57 +1,35 @@
-import { Position } from "@blueprintjs/core";
-import { useMemo } from "react";
-import RawMoreDropdown from "src/components/common/MoreDropdown";
-import { Props as ChooserProps } from "src/components/DropboxChooser";
-import Menu from "./components/Menu";
-import { Collection } from "src/common/entities";
+import { Fragment } from "react";
+import { Props } from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/types";
+import {
+  DATASET_MORE_BUTTON,
+  MORE_ICON_BUTTON_PROPS,
+} from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/constants";
+import { StyledButton } from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/style";
+import { useMoreMenu } from "src/views/Collection/hooks/useMoreMenu";
+import Menu from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/components/Menu";
 
-interface Props {
-  collectionId: Collection["id"];
-  datasetId?: string;
-  isPublished: boolean;
-  revisionsEnabled: boolean;
-  onUploadFile: ChooserProps["onUploadFile"];
-  isLoading: boolean;
-  disabled: boolean;
-}
-
-const MoreDropdown = ({
+export default function MoreDropdown({
```

Comment:
Complete overhaul of dataset row `MoreDropdown` component!

- changed FC to preferred "regular" style,
- grouped props by use,
- use of supporting files such as types.ts as per WMG code style, and
- replacement of BlueprintJS components with SDS components.